keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifact/ArtifactDeployedListener.kt

```diff
@@ -0,0 +1,44 @@
+package com.netflix.spinnaker.keel.artifact
+
+import com.netflix.spinnaker.keel.events.ArtifactVersionDeployed
+import com.netflix.spinnaker.keel.persistence.KeelRepository
+import kotlinx.coroutines.runBlocking
+import org.slf4j.LoggerFactory
+import org.springframework.context.event.EventListener
+import org.springframework.stereotype.Component
+
+@Component
+class ArtifactDeployedListener(
+  private val repository: KeelRepository
+) {
+  private val log = LoggerFactory.getLogger(javaClass)
+
+  @EventListener(ArtifactVersionDeployed::class)
```

Comment:
Trying to understand the logic here - `onArtifactVersionDeployed` is called when an `ArtifactVersionDeployed` event is emitted, correct? If this event has published, it means that the artifact is deployed? if so, why the call to `isApprovedFor` is done after it being deployed?

Reply:
Great question. I had the same question myself as I was porting this logic over!

The event is emitted when we check resources and get the current server group from clouddriver. We emit this event for the version we see in that current server group. I thought it might be good to keep this logic so that we would essentially mark the version as approved then deployed as we went through pushing the artifact through the environments. I think this logic avoids us marking an artifact as `DEPLOYED` if it has not met the constraints/etc. 

However, now that I explain it to you I'm wondering if we should take away that check. It seems we'd want to know if an artifact was already in the environment asap. Let me think about this some more.

Reply:
+1 for removing the check. Just as an example, if it had been deployed there outside of keel's control, and the environment in keel has a manual judgement, we can't make any assumptions about whether that constraint actually passed. But we do know it was deployed, so I'd just remove the check and accept that an artifact version may transition directly to deployed.

Reply:
My worry is that our check for whether or not an artifact was approved for an environment right now just checks if a row exists in the `environment_artifact_versions` table, and that the status is not set to veto. So I think changing the logic here would have unintended consequences. I don't want to make a logic change in a PR that is just a clean up.

Reply:
Fair enough. You said you were gonna think about it, so I wasn't sure what you had decided. Fine by me. 😃